gadgets/snapshot_socket/program.bpf.c

```diff
@@ -112,7 +129,7 @@ socket_bpf_seq_write(struct seq_file *seq, __u16 family, __u16 proto,
 	entry.src.proto = entry.dst.proto = proto;
 	entry.src.port = bpf_htons(srcp);
 	entry.dst.port = bpf_htons(destp);
-	entry.state = state;
+	entry.state_raw = state_raw;
 	entry.ino = ino;
 	entry.netns_id = netns;
 
```

Comment:
I would suggest handling here the case for UDP. Something like this could work:

```c
	// Transform state into something more suitable for UDP
	if (proto == IPPROTO_UDP) {
		switch (state_raw) {
		case ESTABLISHED:
			entry.state = ACTIVE;
			break;
		case CLOSE:
			entry.state = INACTIVE;
			break;
		default:
			entry.state = UNKNOWN;
			break;
		}
	}
```